src/Functions/FunctionBase64Conversion.h

```diff
@@ -85,6 +85,85 @@ struct TryBase64Decode
     }
 };
 
+struct Base64UrlEncode : Base64Encode
+{
+    static constexpr auto name = "base64UrlEncode";
+
+    static size_t perform(const std::span<const UInt8> src, UInt8 * dst)
+    {
+        auto out_len = Base64Encode::perform(src, dst);
+
+        // Do postprocessing as described in https://datatracker.ietf.org/doc/html/rfc4648#page-7
+        for (size_t i = 0; i < out_len; ++i)
+        {
+            switch (dst[i])
+            {
+            case '/':
+                dst[i] = '_';
+                break;
+            case '+':
+                dst[i] = '-';
+                break;
+            case '=': // stop when padding is detected
+                return i;
+            default:
+                break;
+            }
+        }
+        return out_len;
+    }
+};
+
+struct Base64UrlDecode : Base64Decode
+{
+    static constexpr auto name = "base64UrlDecode";
+
+    static size_t perform(const std::span<const UInt8> src, UInt8 * dst)
+    {
+        std::vector<UInt8> tmp{};
+        // insert padding to please alcomp library
+        auto size = src.size();
+        auto remainder = size % 4;
```

Comment:
`size_t remainder` (the type is trivial in this case but as a general rule, `auto` shall only be used when the right side makes the type immediately clear, e.g. `auto x = static_cast<SomeThing>(y)`.